plugins/acl/static-role-based/src/main/java/org/apache/cloudstack/acl/StaticRoleBasedAPIAccessChecker.java

```diff
@@ -107,6 +107,10 @@ public boolean checkAccess(User user, String commandName) throws PermissionDenie
 
     @Override
     public boolean checkAccess(Account account, String commandName) {
+        if (isEnabled()) {
+            return true;
+        }
```

Comment:
if this prevents the escalation talked about fine, but it seems illogical to return true to prevent access. would return false not make more sense?

Reply:
@DaanHoogland I don't think so. Actually this `isEnabled` method checks if `DynamicRoleBasedAPIAccessChecker` is enabled. Returning true here is just passing the check responsibility to another class.

Reply:
yes @hsato03 I think you are right. I consider it a bug that the StaticRoleBasedAPIAccessChecker.isEnabled() would return true if it is disabled (but the dynamic one is enabled). Maybe this is a discussion out of scope for your change. As said, "if this prevents the escalation talked about (it is) fine"

Reply:
yes, it looks like a bug @DaanHoogland 
we should change the return value of `isEnabled` in this class

Reply:
cc @rohityadavcloud ^